src/debugger/gdb-stub.c

```diff
@@ -433,7 +453,35 @@ static void _processQSupportedCommand(struct GDBStub* stub, const char* message)
 		}
 		message = end + 1;
 	}
-	strncpy(stub->outgoing, "swbreak+;hwbreak+", GDB_STUB_MAX_LINE - 4);
+	strncpy(stub->outgoing, "swbreak+;hwbreak+;qXfer:features:read+;qXfer:memory-map:read+", GDB_STUB_MAX_LINE - 4);
+}
+
+static void _processQXferCommand(struct GDBStub* stub, const char* params, const char* data) {
+	unsigned int offset = 0;
+	unsigned int length = 0;
+
+	unsigned int index = 0;
+	while (params[index] != ',') {
+		offset <<= 4;
+		offset |= _hex2int(&params[index], 1);
+		index++;
+	}
```

Comment:
```suggestion
	for (index = 0; params[index] != ','; ++index) {
		offset <<= 4;
		offset |= _hex2int(&params[index], 1);
	}
```

Reply:
Also there may already be a function for constructing these hex numbers automatically. I would need to check.